src/main/java/com/sforce/cd/apexUnit/client/utils/ApexClassFetcherUtils.java

```diff
@@ -321,8 +322,8 @@ public static String fetchAndAddToMapApexClassIdBasedOnName(PartnerConnection co
 		}
 		if (queryResult != null && queryResult.getDone()) {
 			String[] classIds = fetchApexClassesAsArray(queryResult);
-			for (String classId : classIds) {
-				apexClassId = classId;
+			if(classIds.length>0){
```

Comment:
Hey @adarsh-ramakrishna-sfdc ,

Yes, creating an empty object outputs
queryResult.getDone() as false 
But even then getRecords is empty but not null.